### PR TITLE
Suppress categorical warning (fixes #3379)

### LINKED ISSUE #3379: [python] suppress the warning about categorical feature override
```
C:\ProgramData\Anaconda3\lib\site-packages\lightgbm\basic.py:1555: UserWarning: categorical_feature in Dataset is overridden.
New categorical_feature is ['C01', 'C02', 'C03', 'C04', 'C05', 'C06', 'C07', 'C08', 'C09', 'C10', 'C11', 'C12', 'C13', 'C14', 'C15', 'C16', 'C17', 'C18', 'C19', 'C20', 'C21', 'C22', 'C23', 'C24', 'C25', 'C26']
  'New categorical_feature is {}'.format(sorted(list(categorical_feature))))
C:\ProgramData\Anaconda3\lib\site-packages\lightgbm\basic.py:1286: UserWarning: Overriding the parameters from Reference Dataset.
  warnings.warn('Overriding the parameters from Reference Dataset.')
C:\ProgramData\Anaconda3\lib\site-packages\lightgbm\basic.py:1098: UserWarning: categorical_column in param dict is overridden.
  warnings.warn('{} in param dict is overridden.'.format(cat_alias))
```
`categorical_column` could be set in both `lgb.train` and `lgb.Dataset` . But this warning seems always show up if setting  `categorical_column`. I think this is quite annoying.

### PR DESCRIPTION
**Pull request by hzy46**
To fix #3379

### CODE REVIEW

**Comment by jameslamb** [comment]
I just added `(fixes #3379)` to the PR description. I try to do this whenever I open a PR that fixes another issue. PR titles become items in the release notes (e.g. https://github.com/microsoft/LightGBM/releases/tag/v3.3.1), and having the `fixes #xxxx` there makes it easy for users who want to learn more about a particular item in the release notes to click through to the underlying issue.

**Comment by jameslamb** [comment]
oh! Also @hzy46 please check your `~/.gitconfig`. It looks like the commits you've pushed here aren't tied to your GitHub account.


Notice that your profile picture is not showing up:

image

See my suggestions in https://github.com/microsoft/LightGBM/pull/3607#issuecomment-735358654 and https://github.com/uptake/pkgnet/pull/284#pullrequestreview-518230330 for how to fix this.

File: python-package/lightgbm/basic.py [hunk 1]
```diff
@@ -1776,8 +1799,10 @@ def construct(self):
         if self.handle is None:
             if self.reference is not None:
                 reference_params = self.reference.get_params()
-                if self.get_params() != reference_params:
-                    _log_warning('Overriding the parameters from Reference Dataset.')
+                params = self.get_params()
+                if params != reference_params:
+                    if self._compare_params_for_warning(params, reference_params) is False:
```
Comment (shiyu1994) [inline python-package/lightgbm/basic.py:1804]: According to `get_params`, both `params` and `reference_params` can be `None`. In this case they cannot be directly iterated in the `_compare_params_for_warning` method.

Comment (hzy46) [reply python-package/lightgbm/basic.py:1804]: Thanks shiyu. I also found this issue in the CI. Will add a check on this.

**Comment by hzy46** [comment]
Close this to use another PR.

**Comment by hzy46** [comment]
Thanks @jameslamb . I used a different PR. Now the author info issue should be addressed in #4768

**Comment by jameslamb** [comment]
> I used a different PR. Now the author info issue should be addressed in

Thanks! It looks like you already did `git reset`, squashed the commits from this PR, and force-pushed them here so another PR wasn't necessary. I can see that the commit on this PR is tied to your user.

image

But we can review the new PR and keep this, no problem! Just letting you know in case you have this problem in the future.

**Pull request closed without merging**